sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncContainer.java

```diff
@@ -1087,6 +1096,131 @@ ItemDeserializer getItemDeserializer() {
+     * </p>
+     * <p>
+     *     We can get the corresponding operation using {@link CosmosBulkOperationResponse#getOperation()} and
+     *     it's response using {@link CosmosBulkOperationResponse#getResponse()}. If the operation was executed
+     *     successfully, the value returned by {@link com.azure.cosmos.models.CosmosBulkItemResponse#isSuccessStatusCode()} will be true. To get
+     *     actual status use {@link com.azure.cosmos.models.CosmosBulkItemResponse#getStatusCode()}.
+     * </p>
+     * To check if the operation had any exception, use {@link CosmosBulkOperationResponse#getException()} to
+     * get the exception.
+     */
+    @Beta(value = Beta.SinceVersion.V1, warningText =
+        Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public <TContext> Flux<CosmosBulkOperationResponse<TContext>> executeBulkOperations(
+        Flux<CosmosItemOperation> operations,
+        CosmosBulkExecutionOptions bulkOptions) {
+        if (bulkOptions == null) {
+            bulkOptions = new CosmosBulkExecutionOptions();
+        }
+
+        Flux<CosmosItemOperation> operationFlux = operations.flatMap(cosmosItemOperation -> {
+            Mono<CosmosItemOperation> cosmosItemOperationMono;
+            if (cosmosItemOperation.getItem() != null) {
+                ObjectNode objectNode =
+                    EncryptionUtils.getSimpleObjectMapper().valueToTree(cosmosItemOperation.getItem());
+                assert cosmosItemOperation instanceof ItemBulkOperation;
+                cosmosItemOperationMono =
+                    this.encryptionProcessor.encryptObjectNode(objectNode).map(encryptedItem -> new ItemBulkOperation<>(
+                        cosmosItemOperation.getOperationType(),
+                        cosmosItemOperation.getId(),
+                        cosmosItemOperation.getPartitionKeyValue(),
+                        ((ItemBulkOperation<JsonNode, TContext>) cosmosItemOperation).getRequestOptions(),
+                        encryptedItem,
+                        cosmosItemOperation.getContext()
+                    ));
+            } else {
+                cosmosItemOperationMono = Mono.just(
+                    new ItemBulkOperation<>(
+                        cosmosItemOperation.getOperationType(),
+                        cosmosItemOperation.getId(),
+                        cosmosItemOperation.getPartitionKeyValue(),
+                        ((ItemBulkOperation<JsonNode, TContext>) cosmosItemOperation).getRequestOptions(),
+                        null,
+                        cosmosItemOperation.getContext()
+                    )
+                );
+            }
+            return cosmosItemOperationMono;
+        });
+
+        final CosmosBulkExecutionOptions cosmosBulkExecutionOptions = bulkOptions;
+        Mono<Flux<CosmosItemOperation>> then = operationFlux.then(Mono.defer(() -> {
```

Comment:
This look weird , we are creating mono of flux , please check if this can be avoided.
I think you don't need this mono , all you are doing is calling setRequestHeaders(cosmosBulkExecutionOptions);

Reply:
Fixed the code, with only having the below code to set the request headers -> "setRequestHeaders(cosmosBulkExecutionOptions)"

Reply:
This will not work, because `encryptionProcessor` is not initialized if the encryption does not happen. Encryption happens only when the above flux gets executed. And this header code will get executed before encryption is happening. 
This header needs to be set in the flux chain. Please revert this code to its original state.